### ISSUE TITLE
latest-armv7 removed

### ISSUE THREAD

**Issue by esticle** [manios/docker-nagios#18]
Hi,

I was just moving my Nagios container to another RPI and noticed:

```
$ docker pull manios/nagios:latest-armv7
Error response from daemon: manifest for manios/nagios:latest-armv7 not found: manifest unknown: manifest unknown
```
Any idea what gives? It still appears on the [hub page](https://hub.docker.com/r/manios/nagios/builds)

**Comment by manios** [comment]
Hi @esticle ,

There is no tag named `latest-armv7` as it is not needed anymore. I have created a multiplatform tag `latest`. So if you pull with:

```bash
docker pull manios/nagios:latest
```

in a Raspberry Pi 3, it will automatically pull the `arm/v7` image variant. **However sometimes**, because docker manifest related features are still experimental (after 2+ years of their introduction) it has happened when I tested in Raspberry Pi 1 (arm-v6) and Raspberry Pi 3 (arm-v7) that it does not download the arm image but the amd64 .

Thus I had to define explicitly :

```bash
docker pull --platform=linux/arm/v6 manios/nagios
# or
docker pull --platform=linux/arm/v7 manios/nagios
```

and it worked.

Does this help you?
Best regards,  
Christos

**Comment by esticle** [comment]
Thanks Christos,

That does help indeed, thanks!  I'm still finding a x86 compiled version though when I pull linux/arm/v7:

```
$ docker pull --platform=linux/arm/v7 manios/nagios
Using default tag: latest
latest: Pulling from manios/nagios
Digest: sha256:56121d4c11d6bbffd5afef3dcb148460ed1b8d9f522879e20bd080799ffcb3f6
Status: Image is up to date for manios/nagios:latest
docker.io/manios/nagios:latest
$ docker image ls -a | grep nagios
manios/nagios                  latest              0a4f714201b8        6 months ago        99.7MB
$ 
$ docker exec -it nagios bash
bash-5.0# apk add file
(1/2) Installing libmagic (5.37-r1)
(2/2) Installing file (5.37-r1)
Executing busybox-1.31.1-r9.trigger
OK: 140 MiB in 103 packages
bash-5.0# file /opt/nagios/bin/nagios
/opt/nagios/bin/nagios: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-x86_64.so.1, stripped
bash-5.0# /opt/nagios/bin/nagios
bash: /opt/nagios/bin/nagios: cannot execute binary file: Exec format error
bash-5.0# ldd !$
ldd /opt/nagios/bin/nagios
/lib/ld-musl-armhf.so.1: /opt/nagios/bin/nagios: Not a valid dynamic program
bash-5.0# uname -a
Linux 95e2625a1f50 4.19.66-v7+ #1253 SMP Thu Aug 15 11:49:46 BST 2019 armv7l Linux
bash-5.0# 
```
Not sure what else I can check..

Thanks!
E

**Comment by manios** [comment]
Hi @esticle ,

After investigating your issue, I found that the `linux/arm/v6` and `linux/arm/v7` images were downloading the wrong platform version of [gosu](https://github.com/tianon/gosu) binary and probably that's why you have seen these errors. I have fixed the image tags `latest` and `4.4.5` . I have already tested it in a Raspberry Pi 3 and in an `amd64` PC. Thanks for spotting that!

Can you pull and test again? 

Best regards,
Christos

**Comment by esticle** [comment]
> Hi @esticle ,
> 
> After investigating your issue, I found that the `linux/arm/v6` and `linux/arm/v7` images were downloading the wrong platform version of [gosu](https://github.com/tianon/gosu) binary and probably that's why you have seen these errors.

That explains it, thanks!

Confirmed all is good on Arm v7.

Appreciated!
